### PR TITLE
Survey

### PR DESCRIPTION
**Pull request by BenHSmith13**
**Description**

Makes reflection type questions work

### CODE REVIEW

File: client/js/_author/components/assessments/question_types/_question.jsx [hunk 1]
```diff
@@ -55,9 +55,27 @@ export default class Question extends React.Component {
...
+    if (reflect) {
+      type = item.multipleAnswer ? 'multipleReflection' : 'reflection';
+    }
+
+    this.props.updateItem({
+      id: item.id,
+      type,
+      question: {
+        type,
+        choices: true,
```
Comment (nick-benoit14) [inline client/js/_author/components/assessments/question_types/_question.jsx]: Choices gets set to true for reflection types?

File: client/js/_author/components/assessments/question_types/multiple_choice_option.jsx [hunk 2]
```diff
@@ -1,21 +1,26 @@
 import React      from 'react';
+import Radio      from './question_common/option_radio';
 
 export default function multipleChoiceOptions(props) {
+
+  const selector = () => {
+    // TODO: this is where I will put checkboxes
+    if (props.itemType === 'multipleChoice') {
```
Comment (nick-benoit14) [inline client/js/_author/components/assessments/question_types/multiple_choice_option.jsx]: Should we define these as constants somewhere?

File: client/js/_author/components/assessments/question_types/_question.jsx (hunk 1 above)
Comment (BenHSmith13) [inline client/js/_author/components/assessments/question_types/_question.jsx]: No, but when we switch to reflection we need to update  values on the choice fields, which only happens if you pass in choices.

File: client/js/_author/components/assessments/question_types/multiple_choice_option.jsx (hunk 2 above)
Comment (BenHSmith13) [inline client/js/_author/components/assessments/question_types/multiple_choice_option.jsx]: probably

File: client/js/constants/language_types.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
+export const languages = {
+  english: '639-2%3AENG%40ISO',
+};
+
+export function getLanguage(langCode) {
+  switch (langCode) {
```
Comment (nick-benoit14) [inline client/js/constants/language_types.js]: Should we use something like we are doing in `genus_types.js`
```
return _.findKey(types.item, (genusType) => type === genusType) || null;
```

File: client/js/middleware/serialization/qbank/serializers/survey.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import { scrub }      from '../../serializer_utils';
+import genusTypes     from '../../../../constants/genus_types';
+import guid           from '../../../../utils/guid';
+
+function serializeChoices(originalChoices, newChoiceAttributes) {
+  const choices = _.map(originalChoices, (choice) => {
+    const updateValues = newChoiceAttributes[choice.id];
+    return {
+      id: choice.id,
+      text: _.get(updateValues, 'text') || choice.text,
```
Comment (nick-benoit14) [inline client/js/middleware/serialization/qbank/serializers/survey.jsx]: Does `_.get` do anything if you are only looking one level deep?

File: client/js/middleware/serialization/qbank/serializers/survey.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  if (newQuestionAttributes.choices) {
+    newQuestion.choices = serializeChoices(originalQuestion.choices, newQuestionAttributes.choices);
+  }
+
+  return scrub(newQuestion);
+}
+
+function serializeAnswers(originalChoices, newChoiceAttributes) {
+  let newFeedback = null;
+  _.forEach(newChoiceAttributes, (choice) => {
```
Comment (nick-benoit14) [inline client/js/middleware/serialization/qbank/serializers/survey.jsx]: Should this get replace with `_.find`?

File: client/js/middleware/serialization/qbank/serializers/survey.jsx (hunk 4 above)
Comment (BenHSmith13) [inline client/js/middleware/serialization/qbank/serializers/survey.jsx]: updateValues may be null/undefined, and _.get won't break on that

File: client/js/middleware/serialization/qbank/serializers/survey.jsx (hunk 5 above)
Comment (BenHSmith13) [inline client/js/middleware/serialization/qbank/serializers/survey.jsx]: I guess, but not really.

Comment (BenHSmith13) [inline client/js/middleware/serialization/qbank/serializers/survey.jsx]: All the feedback need to be the same, so we go through all the choices, and if one of them has feedback, all of them need that same feedback. I could do `const thing = _.find(newChoiceAttributes, choice => !!choice.feedback);`  `const newFeedBack = thing[0].feedback;` Either way I guess

File: client/js/middleware/serialization/qbank/serializers/survey.jsx (hunk 4 above)
Comment (nick-benoit14) [inline client/js/middleware/serialization/qbank/serializers/survey.jsx]: Oh cool, I didn't realize it worked if the first parameter is null. Is it worth refactoring to use `.get(updateValues, 'text', choice.text)` to make use of the default value parameter

Comment (BenHSmith13) [inline client/js/middleware/serialization/qbank/serializers/survey.jsx]: Sure

File: client/js/constants/language_types.js (hunk 3 above)
Comment (BenHSmith13) [inline client/js/constants/language_types.js]: yup

File: client/js/_author/components/assessments/question_types/multiple_choice_option.jsx (hunk 2 above)
Comment (BenHSmith13) [inline client/js/_author/components/assessments/question_types/multiple_choice_option.jsx]: lets do that in another PR, it should really be an all at once kind of thing

File: client/js/_author/components/assessments/question_types/_question.jsx (hunk 1 above)
Comment (dspencer001) [inline client/js/_author/components/assessments/question_types/_question.jsx]: Add a comment indicating this

**Comment by dspencer001** [comment]
LGTM

File: client/js/_author/components/assessments/question_types/_question.jsx (hunk 1 above)
Comment (BenHSmith13) [inline client/js/_author/components/assessments/question_types/_question.jsx]: done
